core/membuffer.cpp

```diff
@@ -66,7 +76,7 @@ void make_room(struct membuffer *b, unsigned int size)
 		char *n;
 		/* round it up to not reallocate all the time.. */
 		needed = needed * 9 / 8 + 1024;
-		n = realloc(b->buffer, needed);
+		n = (char *)realloc(b->buffer, needed);
```

Comment:
Christ. This reminds me why I hate C++ so much. Stupid pointless casts because the C++ people screwed up "void *".

Oh well. Everything else looks fine.